### PR TITLE
reduce dependabot frequency

### PR DESCRIPTION
**Pull request by rbren**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

no changelog needed

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

We currently get inundated with dependabot PRs. Often it's pulling in a single patch at a time.

~~By slowing down dbot, we end up making bigger version jumps less frequently, with fewer overall PRs.~~

~~There's a tradeoff here--we'll spend longer with missing patch updates. I _think_ dbot is smart enough to send us security fixes first though.~~

~~I've also slowed dbot WAY down on the docs/ folder, since that's much less of a security threat.~~

Edit: I've actually changed this to aggressively group dependency updates. This is maybe too aggressive--one dep breaking will break the whole PR. We'll need to see how often the PR has failing tests...I've isolated packages that tend to break the tests.

The changes should leave us with 2 dbot PRs per day (one FE, one BE) with most of the available updates grouped together. We'll get a 3rd PR on wednesdays for the docs/ folder (which is much less of a problem since it's just docs)

---
**Link of any specific issues this addresses**

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,38 +1,66 @@
...
+        patterns:
+          - "boto3"
+      security-all:
+        applies-to: "security-updates"
+        patterns:
+          - "*"
+      version-all:
+        applies-to: "version-updates"
+        patterns:
+          - "*"
```
Comment (enyst) [inline .github/dependabot.yml:23]: As far as I recall, from backend packages llama-index and chroma can have breakages that need delay or revert. Is boto3 problematic? Don't remember a case, even though it brings with itself multiple package updates.

Comment (rbren) [reply .github/dependabot.yml:26]: Added! I only vaguely remember a boto3 issue...removed for now

**Review by tobitege — APPROVED** [review]
LGTM

**Comment by enyst** [comment]
@rbren of course this happens 😅, as soon as we're about to rely on dependabot more: https://github.com/All-Hands-AI/OpenHands/issues/4443

**Pull request merged**